numpy/ma/core.py

```diff
@@ -876,7 +876,8 @@ def __call__(self, a, *args, **kwargs):
                 result = self.f(d, *args, **kwargs)
             # Make a mask
             m = ~umath.isfinite(result)
-            m |= self.domain(d)
+            with np.errstate(divide='ignore', invalid='ignore'):
```

Comment:
I'm a bit surprised this is necessary to have `divide='ignore``, and wonder if it suggests a bug. Which `domain` call is the problem? Might it not make more sense to move the `errstate` setting inside that `domain` call (or fix it). 

I'm guessing the `invalid='ignore'` part is explicitly for `NaN`, correct?